java/vector/src/main/codegen/templates/StructWriters.java

```diff
@@ -196,6 +196,15 @@ public void setPosition(int index) {
     }
   }
 
+  <#if mode="Nullable">
+  @Override
+  public void writeNull() {
+    container.setNull(idx());
+    setValueCount(idx()+1);
```

Comment:
is the setValueCount required ?  it should be done after all mutations to the vector are done.

Reply:
same code is present in the other writers too, so ignoring this comment